### PR TITLE
refactor: replace char_u with char 14: remove `STRLEN` part final

### PR DESCRIPTION
**Pull request by dundargoc**
Work on https://github.com/neovim/neovim/issues/459

### CODE REVIEW

**Comment by dundargoc** [comment]
Follow-up to https://github.com/neovim/neovim/pull/20151

**Comment by dundargoc** [comment]
test comment from gh dash

File: src/nvim/runtime.c [hunk 1]
```diff
@@ -106,7 +106,7 @@ void estack_push_ufunc(ufunc_T *ufunc, linenr_T lnum)
 {
   estack_T *entry = estack_push(ETYPE_UFUNC,
                                 (char *)(ufunc->uf_name_exp != NULL
-                                         ? ufunc->uf_name_exp : ufunc->uf_name),
+                                         ? ufunc->uf_name_exp : (char_u *)ufunc->uf_name),
```
Comment (zeertzjq) [inline src/nvim/runtime.c:109]: Better move the `char *` cast in instead.

File: src/nvim/keycodes.c [hunk 2]
```diff
@@ -952,26 +952,26 @@ char *replace_termcodes(const char *const from, const size_t from_len, char **co
     }
 
     if (do_special) {
-      char_u *p, *s, len;
+      char *p, *s, len;
```
Comment (zeertzjq) [inline src/nvim/keycodes.c:955]: PVS may complain about this. Maybe change the type of `len` to `int`.

File: src/nvim/memline.c [hunk 3]
```diff
@@ -679,7 +679,7 @@ static void set_b0_dir_flag(ZERO_BL *b0p, buf_T *buf)
   if (same_directory((char_u *)buf->b_ml.ml_mfp->mf_fname, (char_u *)buf->b_ffname)) {
     b0p->b0_flags |= B0_SAME_DIR;
   } else {
-    b0p->b0_flags &= (uint8_t) ~B0_SAME_DIR;
+    b0p->b0_flags = (char)((uint8_t)b0p->b0_flags & (uint8_t) ~B0_SAME_DIR);
```
Comment (zeertzjq) [inline src/nvim/memline.c:682]: Are so many casts really necessary?

File: src/nvim/memline.c [hunk 4]
```diff
@@ -689,8 +689,8 @@ static void add_b0_fenc(ZERO_BL *b0p, buf_T *buf)
   const int size = B0_FNAME_SIZE_NOCRYPT;
 
   int n = (int)strlen(buf->b_p_fenc);
-  if ((int)STRLEN(b0p->b0_fname) + n + 1 > size) {
-    b0p->b0_flags &= (uint8_t) ~B0_HAS_FENC;
+  if ((int)strlen(b0p->b0_fname) + n + 1 > size) {
+    b0p->b0_flags = (char)((uint8_t)b0p->b0_flags & (uint8_t) ~B0_HAS_FENC);
```
Comment (zeertzjq) [inline src/nvim/memline.c:693]: Are so many casts really necessary?

File: src/nvim/spell.c [hunk 5]
```diff
@@ -925,9 +925,9 @@ bool match_checkcompoundpattern(char_u *ptr, int wlen, garray_T *gap)
   return false;
 }
 
-// Returns true if "flags" is a valid sequence of compound flags and "word"
-// does not have too many syllables.
-bool can_compound(slang_T *slang, const char_u *word, const char_u *flags)
+/// @return  true if "flags" is a valid sequence of compound flags and "word"
+///          does not have too many syllables.
+bool can_compound(slang_T *slang, const char *word, const char *flags)
```
Comment (zeertzjq) [inline src/nvim/spell.c:930]: Changing `flags` to `char *` changes behavior and looks very strange, considering that both `mi_compflags` and `uflags` are `char_u` arrays.

**Pull request merged**